src/main/java/com/metamx/common/parsers/TimestampParser.java

```diff
@@ -93,9 +93,11 @@ public DateTime apply(String input)
         }
       };
     } else if (format.equalsIgnoreCase("nano")) {
-      return new Function<String, DateTime>() {
+      return new Function<String, DateTime>()
```

Comment:
it appears that 4 "else if" for "posix", "ruby", "nano" and "mills" could be replaced by single else if like
```
..
else if (format.equalsIgnoreCase("posix"))  ||
           format.equalsIgnoreCase("ruby"))  ||
           format.equalsIgnoreCase("millis"))  ||
           format.equalsIgnoreCase("nano")) ) {
    return new Function<String, DateTime>() {
        private Function<Number, DateTime> func = createNumericTimestampParser(format);
        public DateTime apply(String input) {
            return func.apply(Long.parseLong(ParserUtils.stripQuotes(input)));
        }
    }
} 
else ...
```